lib/rules/attribute-hyphenation.js

```diff
@@ -94,8 +94,16 @@ module.exports = {
         data: {
           text
         },
-        fix: (fixer) =>
-          fixer.replaceText(node.key, text.replace(name, caseConverter(name)))
+        fix: (fixer) => {
+          if (useHyphenated && text.includes('_')) {
```

Comment:
Probably underscores are also problematic when changing to camelCase, right?

```suggestion
          if (text.includes('_')) {
```

Can you also add test cases for this?